iap_watcher/iap_watcher_test.py

```diff
@@ -103,10 +110,36 @@ def testDisableEnable(self):
         'iap_metadata_key': 'AEF_IAP_state',
         'output_state_file': path,
         'polling_interval': 1,
+        'fetch_keys': False,
       }),
       watcher=self.metadata_watcher_,
       loop_watcher=False)
     self.assertTrue(os.path.isfile(path))
 
+  def testFetchKeys(self):
+    """Tests that keys are fetched when the fetch_keys argument is True."""
+    state_path = '%s/tmp' % self.pathname_
+    key_path = '%s/keys' % self.pathname_
+    self.metadata_watcher_.SetGetMetadataResult('{"enabled": true}')
+    class OsMock:
+      def system(self, cmd):
+        self.cmd = cmd
+    os_mock = OsMock()
+    iap_watcher.Main(Object({
+        'iap_metadata_key': 'AEF_IAP_state',
+        'output_state_file': state_path,
+        'polling_interval': 1,
+        'fetch_keys': True,
+        'output_key_file': key_path,
+      }),
+      watcher=self.metadata_watcher_,
+      loop_watcher=False,
+      os_system=os_mock.system)
```

Comment:
I think you can actually just def mock_os_system() without the class inside this function if you like (and then reference here as os_system=mock_os_system). Up to you.

Reply:
Done, good suggestion!